### PR TITLE
golang client

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

**Comment by ngaut** [comment]
PTAL @siddontang

File: pd-client/rpc.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	msgVersion    uint16 = 1
+	msgMagic      uint16 = 0xdaf4
+)
+
+// The RPC format is header + protocol buffer body
+// Header is 16 bytes, format:
+//  | 0xdaf4(2 bytes magic value) | 0x01(version 2 bytes) | msg_len(4 bytes) | msg_id(8 bytes) |,
+// all use bigendian.
+
+func writeMessage(w io.Writer, msgID uint64, msg proto.Message) error {
```
Comment (siddontang) [inline pd-client/rpc.go]: server has the same function too, extract to a util package.

File: pd-client/rpc_worker.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+			for {
+				select {
+				case req := <-w.requests:
+					pending = append(pending, req)
+				default:
+					break POP_ALL
+				}
+			}
+			if ok := w.handleRequests(pending, readwriter); !ok {
+				goto RECONNECT
```
Comment (siddontang) [inline pd-client/rpc_worker.go]: here will not close conn.

**Comment by disksing** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**